test/clientconn_state_transition_test.go

```diff
@@ -616,3 +617,38 @@ func (s) TestConnectivityStateSubscriber(t *testing.T) {
 		}
 	}
 }
+
+// TestChannelStateWaitingForFirstResolverUpdate verifies the initial
+// state of the channel when a manual name resolver doesn't provide any updates.
+func (s) TestChannelStateWaitingForFirstResolverUpdate(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping test in short mode.")
+	}
+
+	backend := stubserver.StartTestService(t, nil)
+	defer backend.Stop()
+
+	mr := manual.NewBuilderWithScheme("e2e-test")
+	defer mr.Close()
+
+	cc, err := grpc.NewClient(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("Failed to create new client: %v", err)
+	}
+	defer cc.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	cc.Connect()
+
+	testutils.AwaitState(ctx, t, cc, connectivity.Idle)
```

Comment:
This has to be before Connect or else it will race..

Reply:
`connectivity.Idle` moved up before `cc.Connect()`

Reply:
This should go above the context creation and just be an assertion that `State() == IDLE`.  No need to `Await` since it _should_ always be in this state upon creation.